internal/manifests/collector/service_test.go

```diff
@@ -321,6 +321,161 @@ func TestMonitoringService(t *testing.T) {
 	})
 }
 
+func TestExtensionService(t *testing.T) {
```

Comment:
nit... can you change this to be a list test instead here and also assert on the service created to be sure it has the right ports set?

Reply:
Okay, I understood about asserting the ports. But can you elaborate more on `list test instead`? Do you mean something like this:

```
func TestExtensionService(t *testing.T) {
  tests := []struct{
      name.            string
      params          manifests.params
      ports              []corev1.ServicePort
      expectedErr  error
}{
      {
          ....
          ....
      },
      {
          ....
          ....
      },
}

for _,tc := range tests {
    t.Run(tc.name, func (t *testing.T){
        // assertion
    })
}
```

Reply:
yep! you can see an example [here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/internal/manifests/collector/service_test.go#L56-L62) for these.

Reply:
I have added the changes. Kindly review them!

Reply:
@jaronoff97 @iblancasa @swiatekm if the changes looks good, then can we merge this one please?